homeassistant/components/v2c/coordinator.py

```diff
@@ -0,0 +1,45 @@
+"""The v2c component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pytrydan import Trydan, TrydanData
+from pytrydan.exceptions import TrydanError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class V2CUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """DataUpdateCoordinator to gather data from any v2c."""
+
+    def __init__(self, hass: HomeAssistant, evse: Trydan, entry: ConfigEntry) -> None:
+        """Initialize DataUpdateCoordinator for a v2c evse."""
+        self.evse = evse
+        entry_data = entry.data
+        self.entry = entry
```

Comment:
Coordinator already has `self.config_entry` but it's optional, checkout the YouTube coordinator where I overwrite the type